onnxscript/optimizer/_function_folding_test.py

```diff
@@ -9,6 +9,7 @@
 
 
 class FunctionFoldingTest(unittest.TestCase):
+    @unittest.expectedFailure  # FIXME
```

Comment:
@gramalingam It looks like we are missing a few features

Reply:
I checked. This is one of the limitations of the current Identity-elimination ... when the main graph output is computed as `graph-output = Identity(some-value)` ... here we can't replace graph-output by some-value trivially. We should
* use the ir.Value some-value
* rename it to be `graph-output`, and
* update graph-outputs to include this renamed some-value
* And we can do this only if some-value is not a graph-input

For the third step above: I see that `graph.outputs` returns a `list` ... so, I assume it is intended to be mutable? Can we simply assign `graph.outputs[j] = some-value` ?

Reply:
> Can we simply assign graph.outputs[j] = some-value ?

Yes - that works!

Reply:
I the logic you described in `visit_graph`. Please take a look.